Source/src/importers/MaterialImporter.cpp

```diff
@@ -17,67 +17,35 @@ void ColorCopy(const aiColor4D& assimp_color, float4& color)
     color.w = assimp_color.a;
 }
 
-Hachiko::MaterialImporter::MaterialImporter() : Importer(Importer::Type::MATERIAL)
-{
-}
-
 void Hachiko::MaterialImporter::Import(const char* path, YAML::Node& meta)
 {
-    HE_LOG("Entering MaterialImporter: %s", path);
-    YAML::Node material_node = YAML::LoadFile(path);
-    FileSystem::Save(StringUtils::Concat(GetResourcesPreferences()->GetLibraryPath(Resource::Type::MATERIAL),
-        meta[GENERAL_NODE][GENERAL_ID].as<std::string>()).c_str(), material_node);
-}
+    // Only 1 material per material asset will exist
+    static const int resource_index = 0;
+    UID uid = ManageResourceUID(Resource::Type::MATERIAL, resource_index, meta);
 
-void Hachiko::MaterialImporter::ImportWithMeta(const char* path, YAML::Node& meta) 
-{
-    Import(path, meta);
+    YAML::Node material_node = YAML::LoadFile(path);    
+    FileSystem::Save(GetResourcePath(Resource::Type::MATERIAL, uid).c_str(), material_node);
```

Comment:
Could we copy the file into the resource path? just to avoid parsing the yaml

Reply:
Yep that makes sense, changing it 👍